components/SpotifyInterface/SpotifyMakeRequest.c

```diff
@@ -32,222 +42,277 @@ bool Spotify_FindCode(char *Response, uint16_t SizeRes)
+            // Handle HTTP header received, if needed
+            break;
+        case HTTP_EVENT_ON_DATA:
+            ESP_LOGI(TAG, "HTTP_EVENT_ON_DATA, len=%d", evt->data_len);
+            dataToRead = true;
+            if (totalLen + evt->data_len < LONG_BUF)                                    // check if receved data is not larger than buffer size 
+            {                                                                           // set flag true if server set some data
+                memcpy(receivedData + totalLen, evt->data, evt->data_len);              // copy received data to the end of previous received data
+                totalLen += evt->data_len;                                              // update pointer to the end of copied data
             }
-            memset(Response, 0x000, SizeRes);
-            for (uint16_t j = 0; j < sizeof(json); j++)
+            break;
+        case HTTP_EVENT_ON_FINISH:
+            receivedData[totalLen] = '\0';                                              // write 0 to the end of string
+            ESP_LOGI(TAG, "HTTP_EVENT_ON_FINISH");
+            break;
+        case HTTP_EVENT_DISCONNECTED:
+            ESP_LOGI(TAG, "HTTP_EVENT_DISCONNECTED");
+            ESP_LOGI(TAG, "##> %s <##", receivedData); 
+            ESP_LOGW(TAG, "received content length = %d\r\n", totalLen);
+            // if any data received, send a queue so other tasks can receive and process it
+            if (dataToRead == true)                             
             {
-                Response[j] = json[j];
+                if (xQueueSend(httpToSpotifyDataQueue, receivedData, pdMS_TO_TICKS(SEC)) == pdTRUE)
+                {
+                    ESP_LOGI(TAG, "Data sent from HTTP to Spotify Interface Handler");
+                }
+                else
+                {
+                    ESP_LOGE(TAG, "Sending Received Data by Queue failed"); 
+                }
             }
-        }
+            dataToRead = false;     // reset flag tp prepare it for next packets
+            totalLen = 0;           // reset contect length counter
+            break;
+        case HTTP_EVENT_REDIRECT:
+            ESP_LOGI(TAG, "redirected");
+            break;
     }
-    return FlgFindToken;
+    return ESP_OK;
 }
 
 /**
  * @brief This function sends a request to the Spotify login API to exchange an authorization code for an access token.
- * @param[in,out] Buf The character buffer to store the request and receive the response.
- * @param[in] SizeBuf The size of the character buffer.
  * @param[in] code is parameter that we give it before .
  * @param[in] SizeCode The size of the authorization code.
  * @return This function does not return a value.
  */
-void Spotify_SendTokenRequest(char *Buf, size_t SizeBuf, char *code, size_t SizeCode)
-{
+void Spotify_SendTokenRequest(char *code, size_t SizeCode)
+{  
+    esp_http_client_config_t custom_config = {
```

Comment:
If you don't need "SizeCode" , please remove it.

Reply:
done